docs/learn/smart-contract-developers/create-nft-collection-lsp7.md

```diff
@@ -0,0 +1,208 @@
+
+:::info
+
+⌨️ The full code of this example can be found in the 👾 [lukso-hardhat-template](https://github.com/CJ42/LUKSO-Hardhat-template).
+
+:::
+
+:::note
+
+This guide builds on top of a Hardhat project using TypeScript as described in the [Getting Started](../smart-contract-developers/getting-started.md) section.
+
+:::
+
+## Setup
+
+Make sure you have the following dependencies installed:
+
+- [`ethers.js`](https://github.com/ethers-io/ethers.js/) (alternatively you can use [`web3.js`](https://github.com/web3/web3.js))
+- [`@lukso/lsp-smart-contracts`](https://github.com/lukso-network/lsp-smart-contracts/)
+
+```bash
+yarn add ethers @lukso/lsp-smart-contracts
+```
+
+## Create the Smart Contracts
+
+When creating smart contracts representing digital assets on LUKSO, you will need to specify the token type and data keys for the 📄 [LSP4 Digital Asset Metadata](../../standards/tokens/LSP4-Digital-Asset-Metadata) that will be stored in the 🗂️ [ERC725Y](../../standards/lsp-background/erc725.md#erc725y-generic-data-keyvalue-store) storage of the Universal Profile. There are three different token types:
+
+- `0` = Token
+- `1` = NFT
+- `2` = Collection
+
+```solidity title="contracts/TokenTypes.sol"
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+bytes32 constant _LSP4_TOKEN_TYPE_DATA_KEY = 0xe0261fa95db2eb3b5439bd033cda66d56b96f92f243a8228fd87550ed7bdfdb3; // kecca256 hash of the word `LSP4TokenType`
+
+enum TokenType {
+    TOKEN,
+    NFT,
+    COLLECTION
+}
+```
+
+After defining the type of the asset and its 🗂️ [ERC725 data key](../../standards/lsp-background/erc725.md#erc725y-generic-data-keyvalue-store) you can create a custom 🌄 [LSP7 Digital Asset Collection](../../standards/tokens/LSP7-Digital-Asset.md) that extends [LSP7Mintable](../../contracts/contracts/LSP7DigitalAsset/presets/LSP7Mintable.md) so that new assets can be created within the smart contract.
+
+```solidity title="contracts/EventTicketsNFT.sol"
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+// modules
+import {
+    LSP7Mintable
+} from "@lukso/lsp-smart-contracts/contracts/LSP7DigitalAsset/presets/LSP7Mintable.sol";
+
+import {_LSP4_TOKEN_TYPE_DATA_KEY, TokenType} from "../TokenTypes.sol";
+
+contract EventTicketsNFT is LSP7Mintable {
+    constructor(
+        string memory eventName,
+        string memory tokenSymbol,
+        address contractOwner
+    )
+        LSP7Mintable(
+            eventName,
+            tokenSymbol,
+            contractOwner,
+            true // make the token non divisible
+        )
+    {
+        // set the token type
+        _setData(_LSP4_TOKEN_TYPE_DATA_KEY, abi.encode(TokenType.TOKEN));
+    }
+}
+```
+
+Next you define the deployment script.
+
+<Tabs groupId="web3-lib">
+  <TabItem value="ethersjs" label="ethers.js">
+
+<!-- prettier-ignore-start -->
+```js
+import { ethers } from "hardhat";
+
+import {
+    EventTicketsNFT,
+    EventTicketsNFT__factory
+} from "../typechain-types";
```

Comment:
If we use the Typechain types, we would need to the command to generate these types in the setup step above.